lib/src/Base/Func/ExpertMixture.cxx

```diff
@@ -114,17 +126,19 @@ Point ExpertMixture::operator() (const Point & inP) const
   const UnsignedInteger outputDimension = getOutputDimension();
   const UnsignedInteger size = experts_.getSize();
   UnsignedInteger bestClass = 0;
-  // Build the point (x, f(x)) for the first class and grade it according to the classifier
+  // Build the point z for the first class and grade it according to the classifier
   Point mixedPoint(inP);
   Point bestValue(experts_[0](inP));
-  mixedPoint.add(bestValue);
+  if (supervised_)// z=(x, f(x))
+    mixedPoint.add(bestValue);
   Scalar bestGrade = classifier_.grade(mixedPoint, bestClass);
   LOGDEBUG(OSS() << "Class index=" << 0 << ", grade=" << bestGrade << ", value=" << bestValue);
   for (UnsignedInteger classIndex = 1; classIndex < size; ++classIndex)
   {
-    // Build the point (x, f(x)) for each other class and grade it according to the classifier
+    // Build the point z for each other class and grade it according to the classifier
     const Point localValue(experts_[classIndex](inP));
-    for (UnsignedInteger i = 0; i < outputDimension; ++i) mixedPoint[inputDimension + i] = localValue[i];
+    if (supervised_)// z=(x, f(x))
+      std::copy(localValue.begin(), localValue.end(), mixedPoint.begin() + inputDimension);
```

Comment:
Oh, you're right. Though I think it's best to leave as is as this complicates the code.